hy/_compat.py

```diff
@@ -1,6 +1,12 @@
+# fmt: off
```

Comment:
> this defeats the whole purpose of formatting to a consistent style

Of course. That's the point.

In seriousness, there is indeed a long tuple defined inline in this file. Although, the particularly bad choice made by Black in this file is changing

```
    def code_replace(code_obj, **kwargs):
        return type(code_obj)(*(
            kwargs.get(k, getattr(code_obj, k))
            for k in _code_args
        ))
```

to 

```
    def code_replace(code_obj, **kwargs):
        return type(code_obj)(
            *(kwargs.get(k, getattr(code_obj, k)) for k in _code_args)
        )
```

You don't actually think Black's version is more readable, do you? The original version provides clear visual separation between the result and the iterator of the generator expression, and it makes it obvious that the unpacking star applies to the whole generator exprssion, not just the result part.

Reply:
Personally, I do. But that's not the point. The point is for style to be consistent across the entire code base and that means not turning off formatting.

Reply:
Readability isn't the point? Are you saying you would oppose any use of `fmt: off` even if it made the code more readable? Because if so, I don't understand why you're interested in consistency. Consistent formatting is all well and good if it makes the code more readable, but if the result is less readable, it's counteproductive, yes?

You did seem to allow in your previous comment that you might allow `fmt: off` for "defining complex datastructures", which sounds like a readability issue, but you tell me.

Reply:
I mean, yes, I'm puzzled that you would find the second version more reaable, but if you're going to object to any use of `fmt: off` regardless, let's consider that larger issue first.

Reply:
Style is subjective. What you find more readable is different than what I find most readable is different than the next person finds the most readable. The only important metric is consistency in style. Consistency enables forming assumptions and intuitions about the code you're reading. Black's style is good enough and eliminates the need to actively stay consistent entirely. And better still, makes us consistent with the style of numerous other projects in the python ecosystem; which makes jumping between projects even easier. 

Across Pandas, Scikit-learn, and Django you won't find `fmt: off` anywhere. I honestly don't think I've seen it ever used in a project. There's nothing I can think of in the Hy codebase that would merit turning off formatting.

Reply:
> Style is subjective.

Exactly, which is why it can't be trusted to a machine.

> The only important metric is consistency in style.

Imagine if the codebase was golfed to use the absolute minimum whitespace allowed in every case. That would be very consistent. There's a reason that Black goes to elaborate lengths to format things in a somewhat human style instead of just doing that.

> Black's style is good enough

It works fine in most cases and in simple cases, but in that small proportion of more complicated cases that appear in a real codebase, it makes obvious mistakes. I'm not against every single style change Black makes. Many of the changes were fine. I'm just against the insistence on absolute dogmatic adherence to it.

> There's nothing I can think of in the Hy codebase that would merit turning off formatting.

I honestly can't tell if you're categorically against `fmt: off` or if you think readability is a reason to use it, at least sometimes. You seem to be not quite willing to bite the bullet and believe that consistency is all that matters, but you're not willing to entirely discount the importance of readability, either. I can show you a dozen cases where Black did stupid things to our code, but there's no point in that exercise if you just don't care.

Reply:
What are both of your opinions on delimiting individual blocks with `fmt: {off,on}`? Obviously that's more verbose than either a file-wide `fmt: off` or giving up manual formatting entirely, but maybe would it be an acceptable compromise for now? See https://black.readthedocs.io/en/stable/the_black_code_style/current_style.html?highlight=skip#code-style for the relevant docs.

Reply:
One of the reasons I dislike linters is that they oblige codebases to accumulate noisy comments to tell the linter to lay off once in a while. It's better than being locked into autoformatting unconditonally, sure, but that's a moot point so long as Allie will oppose any attempt to turn off autoformatting.